src/csm_testing/tests/sat_functional/nid2xname/test_nid2xname.py

```diff
@@ -30,140 +30,144 @@
 
 class TestNid2Xname(SATTestCase):
 
+    @classmethod
+    def setUpClass(cls):
+        """Set up data used by all test methods."""
+        super().setUpClass()
+        cls.components = cls.get_hsm_components()
+        # Pre-calculate the first node's data for simple tests
+        cls.first_xname = cls.components[0]['Xname']
+        cls.first_nid = cls.components[0]['NID']
+
+    @classmethod
+    def get_hsm_components(cls):
+        """Get component data from HSM."""
+        # Get raw HSM data - no jq processing needed
+        command = "cray hsm state components list --type Node --format json"
+        result = subprocess.run(command, shell=True, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
I'm pretty sure you don't need `shell=True` here anymore since you're not piping into `jq`.

```suggestion
        result = subprocess.run(command, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```